### PR TITLE
Add xclim

### PR DESCRIPTION
**Pull request by Zeitsperre**
`xclim` is a library that extends the functionality of xarray to calculate climate science indicators using dask for multiprocessing support and pint to ensure unit consistency. The list of indicators includes all those previously available through the ICCLIM library as well as many others and is growing weekly. While this library is still in beta, we are aiming for a stable (v1.0) release in the coming months.

Checklist

- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages
- [x] Build number is 0
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

File: recipes/xclim/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,91 @@
+# Note: there are many handy hints in comments in this example -- remove them when you've finalized your recipe
```
Comment (chrisburr) [inline recipes/xclim/meta.yaml]: Can you remove the comments from the example

File: recipes/xclim/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+  # Uncomment the following line if the package is pure Python and the recipe is exactly the same for all platforms.
+  # It is okay if the dependencies are not built for all platforms/versions, although selectors are still not allowed.
+  # See https://conda-forge.org/docs/maintainer/knowledge_base.html#noarch-python for more details.
+  # noarch: python
+  number: 0
+  # If the installation is complex, or different between Unix and Windows, use separate bld.bat and build.sh files instead of this key.
+  # By default, the package will be built for the Python versions supported by conda-forge and for all major OSs.
+  # Add the line "skip: True  # [py<35]" (for example) to limit to Python 3.5 and newer, or "skip: True  # [not win]" to limit to Windows.
+  noarch: python
+  skip: True # [py<35]
```
Comment (chrisburr) [inline recipes/xclim/meta.yaml]: You can't use `skip` with `noarch` but you can constraint it in a different way (see below)
```suggestion
```

File: recipes/xclim/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+  script: "{{ PYTHON }} -m pip install . -vv"
+
+requirements:
+  build:
+    - python
```
Comment (chrisburr) [inline recipes/xclim/meta.yaml]:
```suggestion
    - python >=3.5
```

File: recipes/xclim/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+    - pip
+  run:
+    - python
```
Comment (chrisburr) [inline recipes/xclim/meta.yaml]:
```suggestion
    - python >=3.5
```

File: recipes/xclim/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+    - scipy 1.2.*
```
Comment (chrisburr) [inline recipes/xclim/meta.yaml]: Do these need to be so restrictive? Would something like this be good enough:
```suggestion
    - scipy >=1.2
```

File: recipes/xclim/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+    and can benefit from the parallelization provided by dask. It's objective is to 
+    make it as simple as possible for users to compute indices from large climate 
+    datasets, and for scientists to write new indices with very little boilerplate.
+  doc_url: https://xclim.readthedocs.io
+  dev_url: https://github.com/Ouranosinc/xclim
+
+extra:
+  recipe-maintainers:
+    - Zeitsperre
+    - huard
```
Comment (chrisburr) [inline recipes/xclim/meta.yaml]: @huard Are you happy to be listed as a maintainer of this recipe?

File: recipes/xclim/meta.yaml (hunk 2 above)
Comment (Zeitsperre) [reply recipes/xclim/meta.yaml]: Thanks for the clarification!

File: recipes/xclim/meta.yaml (hunk 5 above)
Comment (Zeitsperre) [reply recipes/xclim/meta.yaml]: I didn't know that that syntax was allowed. Thanks!

File: recipes/xclim/meta.yaml [hunk 7]
```diff
@@ -12,74 +7,48 @@ package:
...
 build:
-  # Uncomment the following line if the package is pure Python and the recipe is exactly the same for all platforms.
-  # It is okay if the dependencies are not built for all platforms/versions, although selectors are still not allowed.
-  # See https://conda-forge.org/docs/maintainer/knowledge_base.html#noarch-python for more details.
-  # noarch: python
-  number: 0
-  # If the installation is complex, or different between Unix and Windows, use separate bld.bat and build.sh files instead of this key.
-  # By default, the package will be built for the Python versions supported by conda-forge and for all major OSs.
-  # Add the line "skip: True  # [py<35]" (for example) to limit to Python 3.5 and newer, or "skip: True  # [not win]" to limit to Windows.
+  number: 1
```
Comment (chrisburr) [inline recipes/xclim/meta.yaml]: You should keep the build number at 0. It only needs to be bumped when the build is deployed.

File: recipes/xclim/meta.yaml [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: 62db7cc3e567ced70cc13132ef4fdfddc470350d0dafc392aec19583af11863d
+
+build:
+  number: 0
+  noarch: python
+  script: "{{ PYTHON }} -m pip install . -vv"
+
+requirements:
+  build:
```
Comment (chrisburr) [inline recipes/xclim/meta.yaml]: These should be under `host` not `build`.
```suggestion
  host:
```
(Pure Python packages rarely need `build` dependencies)

File: recipes/xclim/meta.yaml [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+    - setuptools
+    - pip
+  run:
+    - python >=3.5
+    - scipy >=1.2
+    - numpy >=1.15
+    - pandas >=0.23
+    - cftime >=1.0.3
+    - netCDF4 >=1.4
+    - dask[complete] >=1.2
```
[the remark this answers is not in the document]
Comment (Zeitsperre) [reply recipes/xclim/meta.yaml]: For dask, I believe that adding the [complete] tag adds optional modules. They might not be necessary though. Trying it out.

**Comment by Zeitsperre** [comment]
Not sure where I'm going wrong here. It seems to not be loading setuptools despite being listed under `host`. Is there a special syntax for ensuring that setuptools are loaded prior to running pip?

File: recipes/xclim/meta.yaml [hunk 10]
```diff
@@ -0,0 +1,58 @@
...
+build:
+  number: 0
+  noarch: python
+  script: "{{ PYTHON }} -m pip install . -vv"
+
+requirements:
+  host:
+    - python >=3.5
+    - setuptools
+    - pip
```
Comment (chrisburr) [inline recipes/xclim/meta.yaml]: The error message is always a little confusing but at the far right it states that it is completing that `pytest-runner` is missing:
```suggestion
    - pip
    - pytest-runner
```

**Pull request merged**